packages/gamut/src/ToolTip/index.tsx

```diff
@@ -1,18 +1,156 @@
+
+const arrowOffsetHorizontal = `calc(${arrowPaddingHorizontal} - (${arrowWidth} / 2))`;
+const arrowOffsetVertical = `calc(-${arrowHeight} / 2)`;
+
+const slightShadow = `rgba(0, 0, 0, 0.15)`;
+
+const TooltipWrapper = styled.div`
+  position: relative;
+  display: inline-flex;
+`;
+
+const TargetContainer = styled.div`
+  cursor: pointer;
+  border: 0;
+  background: none;
+  padding: 0;
+`;
+
+type ToolTipContainerProps = {
+  position: ToolTipPosition;
+  variant: VisualTheme;
+};
+
+const ToolTipContainer = styled.div<ToolTipContainerProps>`
+  ${fontSmoothing()}
+
+  display: flex;
+  opacity: 0;
+  transition: opacity 0ms;
+  transition-delay: 750ms;
+  position: absolute;
+  width: 16rem;
+  visibility: hidden;
+
+  // Both before and after psuedo-elements are used because ::after's background should go over the container's
+  // and ::before's box-shadow should be behind the container itself
+  &::after,
+  &::before {
+    content: '';
+    display: block;
+    height: ${arrowHeight};
+    position: absolute;
+    transform: rotate(45deg);
+    width: ${arrowHeight};
+  }
+
+  &::after {
+    ${variant({
+      dark: { backgroundColor: 'black' },
+      light: { backgroundColor: 'white' },
+    })}
+  }
+
+  ${TargetContainer}:hover + &,
+  ${TargetContainer}:focus + &,
+  &:hover {
+    opacity: 1;
+    visibility: visible;
+  }
+
+  ${({ position }) =>
+    ['top-left', 'top-right'].includes(position) &&
+    `
+      bottom: ${containerOffsetVertical};
+
+      &::after,
+      &::before {
+        bottom: ${arrowOffsetVertical};
+      }
+
+      &::before {
+        box-shadow: 2px 2px 4px 0 ${slightShadow};
+      }
+
+      ${ToolTipBody} {
+        box-shadow: 0 2px 4px 0 ${slightShadow};
+      }
+    `}
+
+  ${({ position }) =>
+    ['bottom-left', 'bottom-right'].includes(position) &&
+    `
+      top: ${containerOffsetVertical};
+
+      &::after,
+      &::before {
+        top: ${arrowOffsetVertical};
+      }
+
+      &::before {
+        box-shadow: -2px -2px 4px 0 ${slightShadow};
+      }
+
+      ${ToolTipBody} {
+        box-shadow: 0 0 4px 0 ${slightShadow};
+      }
+    `}
+
+  ${({ position }) =>
+    ['bottom-left', 'top-left'].includes(position) &&
+    `
+      justify-content: flex-end;
+      right: $container-offset-horizontal;
+
+      &::after,
+      &::before {
+        right: ${arrowOffsetHorizontal};
+      }
+    `}
+
+  ${({ position }) =>
+    ['bottom-right', 'top-right'].includes(position) &&
+    `
+      &::after,
+      &::before {
+        left: ${arrowOffsetHorizontal};
+      }
+    `}
+`;
+
+const ToolTipBody = styled.div<{ variant: VisualTheme }>`
+  display: inline-block;
+  font-size: ${pxRem(14)};
+  line-height: ${pxRem(18)};
+  padding: 0.6rem 0.75rem;
+
+  ${variant({
+    dark: { backgroundColor: 'black', textColor: 'white' },
+    light: { backgroundColor: 'white', textColor: 'black' },
+  })}
```

Comment:
Could we update this to use `mode` as the prop name for consistency with other components?  Interim prop bloat but think we're pretty close to being able to make this prop unnecessary and modes fully dynamic :D 

```suggestion
  ${variant({
     prop: 'mode',
     variants: {
    dark: { backgroundColor: 'black', textColor: 'white' },
    light: { backgroundColor: 'white', textColor: 'black' },
  }
  })}
```